src/types/model.ts

```diff
@@ -132,9 +132,12 @@ export class Registry {
     genesisBlockNumber: number
   ): Promise<Registry> {
     const db = storage.getDB();
-    const str = await db.get(
-      getNetworkStorageKey(CONDITIONAL_ORDER_REGISTRY_STORAGE_KEY, network)
-    );
+    const str = await db
+      .get(
+        getNetworkStorageKey(CONDITIONAL_ORDER_REGISTRY_STORAGE_KEY, network)
+      )
+      .then((str) => str)
```

Comment:
Is this needed?

Reply:
In the leveldb implementation, when using `get`, if the requested key doesn't exist, the function will throw a `404` (`LEVEL_NOT_FOUND`). Handling this case is required in the event of database initialisation.

Reply:
Additionally, I've made the error handling here more specific to the initialisation case.

Reply:
I still don't get it.
You are talking about the error handling, but I'm asking about the `then((str) => str)`
If the key doesn't exist, it should not matter the `then` anyway.
Or is that required for some reason?

Also, I would prefer to use try/catch to prevent this sort of confusion, as Leveldb docs seem to prefer as well.

Reply:
I actually removed the `then`.

Reply:
Not [on this PR](https://github.com/cowprotocol/watch-tower/pull/64/files#diff-b43f3ac95c01d64b54dbdfc80288236b116d9fe524c5e03dd74579f4de12edd5R139) 
![image](https://github.com/cowprotocol/watch-tower/assets/43217/65f5ceb3-fe1a-4c57-8b68-cb8c8cbf2f6c)

Unless you mean another PR, which then explains things :)

Reply:
Ohh, sorry, too many PRs :see_no_evil: